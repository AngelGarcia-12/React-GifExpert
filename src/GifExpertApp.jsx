import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);
  const onAddCategory = (newCategory) => {
      if(categories.includes(newCategory)) return;

      setCategories([newCategory, ...categories]);
      //categories.push(newCategory);
      //setCategories(cat => [...categories, 'Persona 5']);
  }

  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>

        {/* Inputs */}
        <AddCategory 
          onNewCategory={onAddCategory}
        />

        {/* Listado de Gif */}
        {categories.map(category => (
            <GifGrid 
              key={category} 
              category={category}/>
          ))
        }
    </>
  )
}
