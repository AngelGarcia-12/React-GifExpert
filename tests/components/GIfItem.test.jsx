import { screen, render, getByRole } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Prueba del componente <GifItem/>', () => { 
    const title = 'saitama';
    const url = 'https://one-punch-man.jpg/';

    test('Debe hacer match con snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
     })

    test('Debe de mostrar la imagen con el URL y el ATL indicado', () => { 
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        //expect(screen.getByRol('img').src).toBe(url);
        //expect(screen.getByRol('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
 })