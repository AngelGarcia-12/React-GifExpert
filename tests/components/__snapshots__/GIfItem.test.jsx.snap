// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifItem/> Debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch-man.jpg/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
